src/screens/DiscoveryScreenV2/elements/RecentSearch.js

```diff
@@ -20,100 +20,106 @@ import { fonts } from '../../../utils/fonts';
  */
 const RecentSearch = (param) => {
-    const {
-        shown = true,
-        setSearchText = () => { },
-        setIsFirstTimeOpen = () => { }
-    } = param
-
-    const [discovery, discoveryDispatch] = React.useContext(Context).discovery
-
-    const [isShown, setIsShown] = React.useState(shown)
-    const [items, setItems] = React.useState(discovery.recentSearch)
-
-    React.useEffect(() => {
-        setIsShown(shown)
-        return () => {
-            setIsShown(false)
-        }
-    }, [shown])
-
-    React.useEffect(() => {
-        setItems(discovery.recentSearch)
-        return () => {
-            setItems([])
-        }
-    }, [discovery.recentSearch])
-
-    const manipulateSearchTermsOrder = async (search) => {
-        const result = await AsyncStorage.getItem(RECENT_SEARCH_TERMS)
-        if (!result) return
-
-        let resultArray = JSON.parse(result)
-        if (resultArray.length <= 1) return
-
-        const itemIndex = resultArray.indexOf(search)
-        resultArray = [search].concat(resultArray)
-        resultArray.splice(itemIndex + 1, 1)
-        DiscoveryAction.setDiscoveryRecentSearch(resultArray, discoveryDispatch)
-        AsyncStorage.setItem(RECENT_SEARCH_TERMS, JSON.stringify(resultArray))
-        setItems(resultArray)
-
-    }
-
-    const fetchDiscoveryData = async (search) => {
-        setSearchText(search)
-        setIsFirstTimeOpen(false)
-        Keyboard.dismiss()
-        manipulateSearchTermsOrder(search)
-    }
-
-    const handleClearRecentSearch = async () => {
-        setIsShown(false)
-        // Remove from context here
-        await AsyncStorage.removeItem(RECENT_SEARCH_TERMS)
-
-        DiscoveryAction.setDiscoveryRecentSearch([], discoveryDispatch)
-    }
-
-    if (isShown && items.length > 0) return <View style={styles.recentContainer}>
+  const {shown = true, setSearchText = () => {}, setIsFirstTimeOpen = () => {}} = param;
+
+  const [discovery, discoveryDispatch] = React.useContext(Context).discovery;
+
+  const [isShown, setIsShown] = React.useState(shown);
+  const [items, setItems] = React.useState(discovery.recentSearch);
+
+  React.useEffect(() => {
+    setIsShown(shown);
+    return () => {
+      setIsShown(false);
+    };
+  }, [shown]);
+
+  React.useEffect(() => {
+    setItems(discovery.recentSearch);
+    return () => {
+      setItems([]);
+    };
+  }, [discovery.recentSearch]);
+
+  const manipulateSearchTermsOrder = async (search) => {
+    const result = await AsyncStorage.getItem(RECENT_SEARCH_TERMS);
+    if (!result) return;
+
+    let resultArray = JSON.parse(result);
+    if (resultArray.length <= 1) return;
+
+    const itemIndex = resultArray.indexOf(search);
+    resultArray = [search].concat(resultArray);
+    resultArray.splice(itemIndex + 1, 1);
+    DiscoveryAction.setDiscoveryRecentSearch(resultArray, discoveryDispatch);
+    AsyncStorage.setItem(RECENT_SEARCH_TERMS, JSON.stringify(resultArray));
+    setItems(resultArray);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There's a potential issue in the `manipulateSearchTermsOrder` function. If the search term is already in the array, it will be duplicated because you're adding it to the start of the array and then removing the original occurrence. However, if the search term is not in the array, it will be added but the original occurrence won't be found and removed, leading to an unnecessary operation. Consider checking if the item exists in the array before manipulating the array.
```diff
  const manipulateSearchTermsOrder = async (search) => {
    const result = await AsyncStorage.getItem(RECENT_SEARCH_TERMS);
    if (!result) return;

    let resultArray = JSON.parse(result);
    if (resultArray.length <= 1) return;

    const itemIndex = resultArray.indexOf(search);
    if (itemIndex !== -1) {
      resultArray.splice(itemIndex, 1);
    }
    resultArray = [search].concat(resultArray);
    DiscoveryAction.setDiscoveryRecentSearch(resultArray, discoveryDispatch);
    AsyncStorage.setItem(RECENT_SEARCH_TERMS, JSON.stringify(resultArray));
    setItems(resultArray);
  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->